src/test/java/org/springframework/data/hazelcast/repository/query/HazelcastPropertyComparatorTest.java

```diff
@@ -0,0 +1,92 @@
+package org.springframework.data.hazelcast.repository.query;
+
+import org.junit.Test;
+
+import java.util.AbstractMap;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toList;
+import static org.junit.Assert.assertEquals;
+
+public class HazelcastPropertyComparatorTest {
+
+    static class TestData {
+        String first;
+        String second;
+        Integer third;
+
+        public TestData(String first, String second, Integer third) {
+            this.first = first;
+            this.second = second;
+            this.third = third;
+        }
+
+        public String getFirst() {
+            return first;
+        }
+
+        public String getSecond() {
+            return second;
+        }
+
+        public Integer getThird() {
+            return third;
+        }
+    }
+
+    @Test
+    public void compareTwoNullProperties() {
+        HazelcastPropertyComparator thirdFieldComparator = new HazelcastPropertyComparator("third", true);
+        TestData testData1 = new TestData("Test", null, null);
+        TestData testData2 = new TestData("Test", null, null);
+        Entry<String, TestData> nullField1 = entryOf("testData1", testData1);
+        Entry<String, TestData> nullField2 = entryOf("testData2", testData2);
+        assertEquals(0, thirdFieldComparator.compare(nullField1, nullField2));
+        assertEquals(0, thirdFieldComparator.compare(nullField2, nullField1));
+    }
+
+    @Test
+    public void compareNullPropertyWithNonNullProperty() {
+        HazelcastPropertyComparator secondFieldComparator = new HazelcastPropertyComparator("second", true);
+        TestData testData1 = new TestData("Test", null, 1);
+        TestData testData2 = new TestData("Test", "Case", 2);
+        Entry<String, TestData> nullField = entryOf("second", testData1);
+        Entry<String, TestData> notNullField = entryOf("second", testData2);
+        assertEquals(1, secondFieldComparator.compare(nullField, notNullField));
+        assertEquals(-1, secondFieldComparator.compare(notNullField, nullField));
+    }
+
+    @Test
+    public void compareOnMultipleProperties() {
```

Comment:
I don't understand this test. Why do you need to create multiple HazelcastPropertyComparator instances and chain them in order to perform unit test? Wouldn't it be possible to simplify it?

Reply:
Yes we can remove it all together , it's the remaining piece of code I used to test some of the lists that were giving us problem.

Reply:
@leszko I was thinking about adding a test for the natural order and another one for when the comparator is used over a complex type property that doesn't implement `Comparable` . With these two additional tests we cover all the remaining possible scenarios.

Reply:
Yeah, that makes sense to me, but I'd keep is simple without aggregating the comparators (like in `comparatorForFields`).